### DISCUSSION TITLE
[Bug]: Exception during Pimcore-Update: " Call to a member function getTable() on null"

### DISCUSSION THREAD

**Discussion by HeicoSindelItShapes** [pimcore/pimcore#11563] [Q&A / Support]
### Expected behavior

Successfull update after "COMPOSER_MEMORY_LIMIT=-1 composer update"

### Actual behavior

[OK] All assets were successfully installed.
[OK] Pimcore data cache cleared successfully
[error] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20211018104331 failed during Execution. Error: "Call to a member function getTable() on null"

### Steps to reproduce

Install Pimcore v. 10.2.8
Use following Bundles
 --- ------------------------------------------ ------------------
      Bundle                                     Method / Error
 --- ------------------------------------------ ------------------
  ✔   FOSJsRoutingBundle                         relative symlink
  ✔   PimcoreCoreBundle                          relative symlink
  ✔   PimcoreAdminBundle                         relative symlink
  ✔   PimcoreDataHubBundle                       relative symlink
  ✔   PimcoreCustomerManagementFrameworkBundle   relative symlink
  ✔   PimcorePerspectiveEditorBundle             relative symlink
  ✔   PimcoreDataImporterBundle                  relative symlink
 --- ------------------------------------------ ------------------

Update via "COMPOSER_MEMORY_LIMIT=-1 composer update"

**Comment by HeicoSindelItShapes** [comment]
It always occurs with the same version for migration (file about 5 versions behind). Even the command "/bin/console pimcore:deployment:classes-rebuild" does not change the result.

**Comment by dvesh3** [comment]
@HeicoSindelItShapes Could please share error line number? as $schema should not be null

**Comment by HeicoSindelItShapes** [comment]
Exit code: 255

  [notice] Migrating up to Pimcore\Bundle\CoreBundle\Migrations\Version20220201132131


  [error] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20211018104331 
  failed during Execution. Error: "Call to a member function getTable() on
  null"

  In EvaluatingGeneratorStrategy.php(54) : eval()'d code line 49:

    Call to a member function getTable() on null

**Comment by dvesh3** [comment]
Sorry, not able to reproduce it. could you please try on latest version 10.3.1? thanks!

**Comment by HeicoSindelItShapes** [comment]
this is what i try to do - i come from 10.2.8 and want do update to 10.3.1

**Comment by HeicoSindelItShapes** [comment]
and pimcore is now on 10.3.1 - although there was an error in the post-update step. What i also can see is, that DataHub seems to work correctly - but at data-importer i only see the import-files but without a icon and i can not open it... maybe it is located somewhere in this area?! could you describe what the update-script do in this step? where it get the schema from - maybe i have i  kind of missconfig

**Comment by HeicoSindelItShapes** [comment]
i have other idea - i can see in the database these tables exist twice: 

object_brick_query_OAuth1Token_12
object_brick_query_OAuth2Token_12
object_brick_store_OAuth1Token_12
object_brick_store_OAuth2Token_12

also as lowercase
object_brick_query_oauth1token_12
object_brick_query_oauth2token_12
object_brick_store_oauth1token_12
object_brick_store_oauth2token_12

as pimcore always create the table in lowercase (as configured in mysql-config) i should delete the other ones? 
does pimcore checks something against this? or could i delete these tables?

**Comment by HeicoSindelItShapes** [comment]
Last update: i have downgraded to 10.2.8 and then step by step forward - on 10.2.8 and 10.2.9 everything works fine - >= 10.3.0 lead to the error above...
